src/zenml/zen_stores/migrations/versions/8ed03137cacc_polymorthic_run_metadata.py

```diff
@@ -0,0 +1,159 @@
+            WHERE pipeline_run_id IS NOT NULL
+            """
+        )
+        session.execute(stmt)
+        stmt = sa.text(
+            """
+            UPDATE run_metadata
+            SET resource_id = step_run_id,
+                resource_type = 'step run'
+            WHERE step_run_id IS NOT NULL
+            """
+        )
+        session.execute(stmt)
+        stmt = sa.text(
+            """
+            UPDATE run_metadata
+            SET resource_id = artifact_id,
+                resource_type = 'artifact'
+            WHERE artifact_id IS NOT NULL
+            """
+        )
+        session.execute(stmt)
+        session.commit()
+
+    with op.batch_alter_table("run_metadata", schema=None) as batch_op:
+        batch_op.alter_column(
+            "resource_type",
+            existing_type=sa.VARCHAR(length=255),
+            existing_nullable=True,
+            nullable=False,
+        )
+        batch_op.alter_column(
+            "resource_id",
+            existing_type=sqlmodel.sql.sqltypes.GUID(),
+            existing_nullable=True,
+            nullable=False,
+        )
+        batch_op.drop_column("step_run_id")
+        batch_op.drop_column("artifact_id")
+        batch_op.drop_column("pipeline_run_id")
+
+    # ### end Alembic commands ###
+
+
+def downgrade() -> None:
+    """Downgrade database schema and/or data back to the previous revision."""
+    # ### commands auto generated by Alembic - please adjust! ###
+    with op.batch_alter_table("run_metadata", schema=None) as batch_op:
+        batch_op.add_column(
+            sa.Column("pipeline_run_id", sa.CHAR(length=32), nullable=True)
+        )
+        batch_op.add_column(
+            sa.Column("artifact_id", sa.CHAR(length=32), nullable=True)
+        )
+        batch_op.add_column(
+            sa.Column("step_run_id", sa.CHAR(length=32), nullable=True)
+        )
+        batch_op.create_foreign_key(
+            "fk_run_metadata_pipeline_run_id_pipeline_run",
+            "pipeline_run",
+            ["pipeline_run_id"],
+            ["id"],
+            ondelete="CASCADE",
+        )
+        batch_op.create_foreign_key(
+            "fk_run_metadata_artifact_id_artifact",
+            "artifact",
+            ["artifact_id"],
+            ["id"],
+            ondelete="CASCADE",
+        )
+        batch_op.create_foreign_key(
+            "fk_run_metadata_step_run_id_step_run",
+            "step_run",
+            ["step_run_id"],
+            ["id"],
+            ondelete="CASCADE",
+        )
+
+    with Session(bind=op.get_bind()) as session:
+        stmt = sa.text(
+            """
+            UPDATE run_metadata
+            SET pipeline_run_id = resource_id
+            WHERE resource_type = 'pipeline run'
+            """
+        )
+        session.execute(stmt)
+        stmt = sa.text(
+            """
+            UPDATE run_metadata
+            SET step_run_id = resource_id
+            WHERE resource_type = 'step run'
```

Comment:
If you change enum naming convention, this would need to change too

Reply:
[b48e13f](https://github.com/zenml-io/zenml/pull/2064/commits/b48e13fc098f5f27cf68a5789079e6a9da69872a)